### PR TITLE
Parse MADT/ACPI table

### PR DESCRIPTION
**Pull request by gsilvis**
This lets us detect other processors, as well as ISA IRQ remappings, and other
things necessary for using the APIC instead of the PIC.  We're not actually
doing anything with this data right now, just printing it at startup.

This is PR 1/4 to enable multiprocessing.  See the branch 'madt-clean' for the full chain of commits.

### CODE REVIEW

File: kernel/apic.c [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    }
+    default:
+      vga_printf("Unimplemented MADT entry with type: %hd\r\n", entry_type);
+      break;
+  }
+  return entry_size;
+}
+
+void apic_init (void) {
+  vga_printf("CPU-Local APIC is at 0x%X\r\n", madt->local_interrupt_controller_address);
```
Comment (cstanfill) [inline kernel/apic.c]: it doesn't look like you're reading from local_interrupt_controller_address to get the apic, what does this actually point to?

File: kernel/apic.c [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+/* Returns the size of the entry */
+static uint8_t analyze_madt_entry (uint8_t *entry) {
+  uint16_t *entry_16 = (uint16_t *)entry;
+  uint32_t *entry_32 = (uint32_t *)entry;
+  uint8_t entry_type = entry[0];
+  uint8_t entry_size = entry[1];
+  /* TODO: Print bytes with the hh specifier once it's supported. */
+  switch (entry_type) {
+    case 0:
```
Comment (cstanfill) [inline kernel/apic.c]: These cases should probably be enums

Comment (gsilvis) [reply kernel/apic.c]: Done.

File: kernel/apic.c (hunk 1 above)
Comment (gsilvis) [reply kernel/apic.c]: This is the location of the mmio registers for the local APIC, as opposed to the apic descriptor table [in the madt variable]

Comment (cstanfill) [reply kernel/apic.c]: oh

Comment (gsilvis) [reply kernel/apic.c]: In the next CL, I'll save it into a variable named 'apic', if that makes you feel better.

**Pull request merged**